api/base/serializers.py

```diff
@@ -513,8 +513,8 @@ def get_url(self, obj, view_name, request, format):
 
     def to_esi_representation(self, value, envelope='data'):
         relationships = self.to_representation(value)
-        if relationships is not None and 'related' in relationships.keys():
-            href = relationships['related']
+        if relationships is not None and 'links' in relationships.keys() and 'related' in relationships['links'].keys():
```

Comment:
The self link should end up coming back in the request for the relationships.links[].related.href response.